metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/pipeline/persistence/FilePersistenceService.java

```diff
@@ -0,0 +1,116 @@
+
+    private static final Logger FILE_PERSIST_SERVICE_LOGGER = Logger.getLogger(FilePersistenceService.class.getName());
+
+    @Autowired
+    private FileRepository fileRepository;
+
+    @Autowired
+    private SraToAmpt2dConverter<AnalysisFileType, File> fileTypeConverter;
+
+    @Autowired
+    private SraObjectExtractorFromAnalysis sraObjectExtractorFromAnalysis;
+
+    @Override
+    public void run(ApplicationArguments arguments) throws Exception {
+        List<String> analysisAccessions = readAccessionsFromFile(arguments);
+        Map<String, List<AnalysisFileType>> analysisAccessionToFiles = getSraObjectsFromAnalysisDocument(analysisAccessions);
+        List<File> files = convertAnalysisFilesToMetadataFiles(analysisAccessionToFiles);
+        persistFilesToDatabase(files);
+    }
+
+    private Map<String, List<AnalysisFileType>> getSraObjectsFromAnalysisDocument(List<String> accessions) {
+        FILE_PERSIST_SERVICE_LOGGER.log(Level.INFO,
+                "Fetching files from analysis.AnalysisAccessionsCount: " + accessions.size());
+        return sraObjectExtractorFromAnalysis.getSraObjectsFromAnalysis(accessions);
+    }
+
+    private List<File> convertAnalysisFilesToMetadataFiles(Map<String, List<AnalysisFileType>>
+                                                                   analysisAccessionToFiles) {
+        FILE_PERSIST_SERVICE_LOGGER.log(Level.INFO,
+                "Converting AnalysisFiles To Metadata Files.AnalysisCount: " + analysisAccessionToFiles.size());
+        return fileTypeConverter.convertModels(analysisAccessionToFiles.values().stream()
+                .flatMap(l -> l.stream()).collect(Collectors.toList()));
+    }
+
+    private void persistFilesToDatabase(List<File> files) {
+        FILE_PERSIST_SERVICE_LOGGER.log(Level.INFO, "Files count to be persisted : " + files.size());
+        int savedFileCount = 0;
+        for (File file : files) {
+            try {
+                fileRepository.save(file);
+                savedFileCount++;
+            } catch (Exception e) {
+                FILE_PERSIST_SERVICE_LOGGER.log(Level.INFO, "Duplicate File" + file.getHash());
+            }
+        }
+        FILE_PERSIST_SERVICE_LOGGER.log(Level.INFO, "Successfully Persisted Files count : " + savedFileCount);
+    }
+
+    private List<String> readAccessionsFromFile(ApplicationArguments arguments) throws Exception {
+        List<String> analysisAccessionsFilePath = arguments.getOptionValues(ANALYSIS_ACCESSION_FILE_PATH);
+        List<String> analysisAccessions = new ArrayList<>();
+        if (analysisAccessionsFilePath != null) {
+            try {
+                analysisAccessions = Arrays.asList(new String(Files.readAllBytes(Paths.get(getClass().getClassLoader()
+                        .getResource(analysisAccessionsFilePath.get(0)).toURI()))).split("\n"));
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Is it necessary to dump stacktrace here? Anyway is throwing to caller
Same other places